connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -2042,6 +2042,166 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
+  test("SPARK-49121: from_protobuf and to_protobuf SQL functions") {
```

Comment:
@itholic It seems that this test cannot be successfully tested in the Maven daily test, both Java 17 and Java 21：
- Java 21: https://github.com/apache/spark/actions/runs/10509632076/job/29116303271
- Java 17: https://github.com/apache/spark/actions/runs/10508648823/job/29113000899

<img width="897" alt="image" src="https://github.com/user-attachments/assets/e5b8f35d-71c9-4df5-a2b6-21923c7e2853">

also cc  @HyukjinKwon

Reply:
@itholic mind taking a look please?

Reply:
@itholic @HyukjinKwon Attempting to fix: https://github.com/apache/spark/pull/47855

Reply:
Thanks for following up.